pkg/manager/member/tikv_scaler.go

```diff
@@ -41,42 +41,53 @@ func NewTiKVScaler(pdControl pdapi.PDControlInterface,
 	return &tikvScaler{generalScaler{pdControl, pvcLister, pvcControl}, podLister}
 }
 
+func (tsd *tikvScaler) Scale(tc *v1alpha1.TidbCluster, oldSet *apps.StatefulSet, newSet *apps.StatefulSet) error {
+	scaling, _, _, _ := scaleOne(oldSet, newSet)
+	if scaling > 0 {
+		return tsd.ScaleOut(tc, oldSet, newSet)
+	} else if scaling < 0 {
+		return tsd.ScaleIn(tc, oldSet, newSet)
+	}
+	return nil
+}
+
 func (tsd *tikvScaler) ScaleOut(tc *v1alpha1.TidbCluster, oldSet *apps.StatefulSet, newSet *apps.StatefulSet) error {
+	_, ordinal, replicas, deleteSlots := scaleOne(oldSet, newSet)
```

Comment:
recalculating again because of less code needs to be changed (otherwise ScaleOut/ScaleIn function signatures need to change and a lot of tests must be updated)